.github/workflows/container.yml

```diff
@@ -0,0 +1,114 @@
+name: Docker Security Checks
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      
+      # Rule #0: Keep Host and Docker Up to Date
+      - name: Check Docker Version
+        run: |
+          docker --version
+          # Optionally, add checks against a list of allowed versions
+
+      # Rule #1: Do Not Expose the Docker Daemon Socket
+      - name: Check for Docker Socket Exposure
+        run: |
+          if grep -q "/var/run/docker.sock" **/*.yml; then
+            echo "Error: Docker socket is exposed."
+            exit 1
+          fi
+
+      # Rule #2: Set a User
+      - name: Run Hadolint for Dockerfile User Check
+        uses: hadolint/hadolint-action@v1
+        with:
+          args: '--ignore DL3006' # Ignore user directive warning
+
+      # Rule #3: Limit Capabilities
+      - name: Check for Privileged Flags
+        run: |
+          if grep -q '--privileged' **/*.yml; then
+            echo "Error: Privileged flag is used."
+            exit 1
+          fi
+
+      # Rule #4: Prevent In-Container Privilege Escalation
+      - name: Check for Privilege Escalation Prevention
+        run: |
+          if grep -q "allowPrivilegeEscalation: true" **/*.yml; then
+            echo "Error: Privilege escalation is allowed."
+            exit 1
+          fi
+
+      # Rule #5: Be Mindful of Inter-Container Connectivity
+      - name: Check for Custom Network Usage
+        run: |
+          if ! grep -q 'networks:' **/*.yml; then
+            echo "Warning: Custom Docker networks are not used."
+          fi
+
+      # Rule #6: Use Linux Security Module
+      - name: Check for Security Profiles
+        run: |
+          if ! grep -q 'securityOpt:' **/*.yml; then
+            echo "Warning: No security profiles defined."
+          fi
+
+      # Rule #7: Limit Resources
+      - name: Check Resource Limits
+        run: |
+          if ! grep -q 'mem_limit:' **/*.yml; then
+            echo "Warning: No memory limits defined."
+          fi
+
+      # Rule #8: Set Filesystem and Volumes to Read-Only
+      - name: Check for Read-Only Filesystem
+        run: |
+          if grep -q 'read_only: false' **/*.yml; then
+            echo "Error: Read-only filesystem is not set."
+            exit 1
+          fi
```

Comment:
This is checking all yml files in the repo for the existence of read_only: false which I'm pretty sure we don't have. Even if we do, it does not guarantee that our docker container's filesystem has this set. We might want to comment this out for now and come back to it or do a deeper check.